packages/console/src/pages/ApplicationDetails/SamlApplicationDetailsContent/CreateSecretModal.tsx

```diff
@@ -0,0 +1,121 @@
+import { type SamlApplicationSecret } from '@logto/schemas';
+import { addYears, format } from 'date-fns';
+import { useCallback, useEffect, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { toast } from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+import ReactModal from 'react-modal';
+
+import Button from '@/ds-components/Button';
+import DangerousRaw from '@/ds-components/DangerousRaw';
+import FormField from '@/ds-components/FormField';
+import ModalLayout from '@/ds-components/ModalLayout';
+import Select from '@/ds-components/Select';
+import useApi from '@/hooks/use-api';
+import modalStyles from '@/scss/modal.module.scss';
+import { trySubmitSafe } from '@/utils/form';
+
+type FormData = { lifeSpanInYears: string };
+
+type Props = {
+  readonly appId: string;
+  readonly isOpen: boolean;
+  readonly onClose: (createdSamlAppSecret?: SamlApplicationSecret) => void;
+};
+
+const years = Object.freeze([1, 3, 5, 10]);
+
+function CreateSecretModal({ appId, isOpen, onClose }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const {
+    control,
+    watch,
+    formState: { isSubmitting },
+    handleSubmit,
+    reset,
+  } = useForm<FormData>({ defaultValues: { lifeSpanInYears: '1' } });
+  const onCloseHandler = useCallback(
+    (created?: SamlApplicationSecret) => {
+      reset();
+      onClose(created);
+    },
+    [onClose, reset]
+  );
+  const api = useApi();
+  const expirationYears = watch('lifeSpanInYears');
+  const [expirationDate, setExpirationDate] = useState<Date>(addYears(new Date(), 1));
+
+  // Update expiration date every second since our options are relative to the current time (in years).
+  useEffect(() => {
+    const setDate = () => {
+      setExpirationDate(addYears(new Date(), Number(expirationYears)));
```

Comment:
Can we add 1 second to the existing date time instead of initializing a new date object every second? I think the performance should be better that way?

Reply:
The state mentioned here is for display purposes, showing when the certificates/secrets under the current option will expire. The actual expiration time depends on when the certificates/secrets were created.